### DISCUSSION TITLE
heighmap using z-probe

### DISCUSSION THREAD

**Discussion by thecuress** [martin2250/OpenCNCPilot#181] [General]
hello,i am using the probe i put in the picture ( the height is 16mm) ..i do all the procedures .... i scan the part ..i apply the height mapping ...once i launch the program the height of the cutter,it remains 16mm high,and does not go down..... i don't know where i am going wrong.thank you very much for the answer

![Immagine](https://github.com/martin2250/OpenCNCPilot/assets/58488797/f47d8f91-15f0-4336-bc92-509031863376)

**Comment by martin2250** [comment]
Hi, OpenCNCPilot assumes that the tool touches the workpiece directly. So you have to subtract the offset of 16 mm afterwards by moving the coordinate system. You could do that with the following GCode:
```
G0 Z10
G92 Z26
```
It will first move to Z=10 and then tell GRBL that its new Z position is 16mm higher than before.

**Reply by thecuress** [reply]
but do I apply this procedure after giving the heighmap ? tks

**Reply by thecuress** [reply]
perfect works great ...thank you very much for the information
